spec/inputs/xmpp_spec.rb

```diff
@@ -1 +1,106 @@
+# encoding: utf-8
 require "logstash/devutils/rspec/spec_helper"
+require 'logstash/inputs/xmpp'
+
+module Jabber
```

Comment:
Hi,
 as there no official logstash style guide about it, it end up very personal. But in my experience, keeping the test without the definition of mocks and other support classes makes sense as code is simpler and easier to understand then.

I would strongly recommend you to do it as will keep things easy to understand and manage in the long term. But again this is my opinion, i tend to like everything in their place, and for this example mocks belong to their definition files and test belong here, mixing them confuses the read, and somehow makes stuff tied up.

On the other side, even if not official this is a pattern I'm trying to enforce in the hole code base, so it would be nice to have mocks defined in helper files that then gets used in the examples.

Again, is the same example as having smoller methods as possible, we're humans and to understand big pieces of code it gets error prone.